src/redux/reducers/greetingSlice.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = { message: '' };
+
+export const fetchRandomGreeting = createAsyncThunk(
+  'greeting/fetchRandomGreeting',
+  async () => {
+    const response = await fetch('http://localhost:3001/random_greeting');
+    const data = await response.json();
+    return data.message;
+  },
+);
+
+const greetingSlice = createSlice({
+  name: 'greeting',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchRandomGreeting.pending, (state) => {
+        state.message = 'Loading...';
+      })
+      .addCase(fetchRandomGreeting.fulfilled, (state, action) => {
+        state.message = action.payload;
+      })
+      .addCase(fetchRandomGreeting.rejected, (state) => {
+        state.message = 'Error fetching greeting';
+      });
+  },
+});
+
+export default greetingSlice.reducer;
```

Comment:
- [ ] As suggested by the previous reviewer, I kindly suggest that you work on the error seen below as regards fetching the messages.


<img width="260" alt="image" src="https://github.com/MNisarAli/hello-react-front-end/assets/52934128/b46d51f3-1ab9-4f3d-b32a-80bb71d2c81b">

Reply:
Hi @topeogunleye, as you are running the app on port 3001 so it is showing the error. as per the rack-cors config if you please run it on port 3000 it will work fine

Reply:
@MNisarAli, I'm really sorry for my late response. Yes sir, You are absolutely correct. My apologies for any inconveniencies. I was running the react app on port 3001 because the rails backend app is already taking port 3000. 

When I switched to run the backend on port 3001 instead and the frontend on port 3000, it worked:

<img width="268" alt="image" src="https://github.com/MNisarAli/hello-react-front-end/assets/52934128/f98899dc-3c5e-4c1e-8104-214f1f4a1bc5">